### ISSUE TITLE
Add missing tests for Chi_animalSpider and fix broken spider

### ISSUE THREAD

**Issue by jim** [City-Bureau/city-scrapers#181]
[There are only a few tests currently](https://github.com/City-Bureau/documenters-aggregator/blob/master/tests/test_chi_animal.py). We should have better coverage for this spider, something more along the lines of [this](https://github.com/City-Bureau/documenters-aggregator/blob/master/tests/test_chi_pubhealth.py).

**Comment by r-wei** [comment]
@milti want to give this a try? There are 2 things that need to be done:
1.  The spider errors out when you try to run it. Running `scrapy crawl chi_animal` gives the following message:
```  
File "/documenters-aggregator/documenters_aggregator/spiders/chi_animal.py", line 41, in parse
    start_time, start_time_str = self._parse_start(text)
  File "documenters-aggregator/documenters_aggregator/spiders/chi_animal.py", line 115, in _parse_start
    naive_date = dateparse(item)
  File ".virtualenvs/documenters-aggregator/lib/python3.6/site-packages/dateutil/parser.py", line 1182, in parse
    return DEFAULTPARSER.parse(timestr, **kwargs)
  File ".virtualenvs/documenters-aggregator/lib/python3.6/site-packages/dateutil/parser.py", line 559, in parse
    raise ValueError("Unknown string format")
```

2. We should have more tests for this spider. See Jim's comment above. It looks like the [test file we're currently using](https://github.com/City-Bureau/city-scrapers/blob/master/tests/files/chi_animal.html) is from 2017, so we should replace it with the 2018 version. The new version can be downloaded using [curl](http://osxdaily.com/2014/02/13/download-with-curl/).

**Comment by milti** [comment]
the page added an event that doesnt follow the pattern of the other entries:

2018 Meeting Dates:

    March 15, 2018 - 8:30 a.m.
    **May 17, 2018 - 6:30 p.m. - (Evening)**
    July 19, 2018 - 8:30 a.m.
    September 20, 2018 - 8:30 a.m.
    November 15, 2018 - 8:30 a.m.

discussing this with rebecca, we decided in getting the dateparse function to split the line after the am/pm designation and parse the LHS of the split string and dump or contain the RHS in another object.

**Comment by milti** [comment]
adding the current URL to use as a test file has now broken the test: 

=================================== FAILURES ===================================
___________________________________ test_len ___________________________________
    def test_len():
>       assert len(parsed_items) == 3
E       AssertionError: assert 5 == 3
E        +  where 5 = len([{'_type': 'event', 'all_day': False, 'classification': 'Not classified', 'description': None, ...}, {'_type': 'event'...ption': None, ...}, {'_type': 'event', 'all_day': False, 'classification': 'Not classified', 'description': None, ...}])
tests/test_chi_animal.py:13: AssertionError
_______________________________ test_start_time ________________________________
    def test_start_time():
>       assert parsed_items[0]['start_time'] == '2017-09-21T00:00:00-05:00'
E       AssertionError: assert '2018-03-15T08:30:00-05:00' == '2017-09-21T00:00:00-05:00'
E         - 2018-03-15T08:30:00-05:00
E         ?    ^  ^  -  ^ ^
E         + 2017-09-21T00:00:00-05:00
E         ?    ^  ^ +   ^ ^
tests/test_chi_animal.py:21: AssertionError
=================== 2 failed, 12762 passed in 27.98 seconds ====================
The command "invoke runtests" exited with 1.

unsure, BUT corrected the test by changing the assertions for length and start_time.

**Comment by r-wei** [comment]
Hey @milti , it looks like the tests are passing on your fixed-chi-animal branch. Why don't you open a new pull request from milti/fixed-chi-animal to City-Bureau/master, and we can talk there. 

For this issue, we want to add tests to cover all of the fields listed in the [events schema](https://github.com/City-Bureau/city-scrapers/blob/master/docs/06_event_schema.md). Right now, test_chi_animal is only testing name, start_time, sources, timezone. But we want it to test all the fields like classification, location, etc.

**Comment by milti** [comment]
ok, ill look into doing that. 

but tell me, how do you run the tests from the terminal? ive tried from shell CL and it complains about not being able to import pkgs, but i dont know how to run a single test from python shell.

do you have a hint on how to do that? ive tried the *exec(open(<file>).read())* but i dont get any output.

**Comment by r-wei** [comment]
ah, we're using a python package called pytest. If you're in the main repo folder, you can call `pytest tests/test_chi_animal.py` in the terminal.

**Comment by milti** [comment]
thanks, thats the command i was looking for: pytest!

ill keep going on this one.

**Comment by bonfirefan** [comment]
Hey @milti, I think we left you hanging. Hope you're ok with still checking in on this! I think Jim linked a good test file, [test_chi_pubhealth.py](https://github.com/City-Bureau/city-scrapers/blob/master/tests/test_chi_pubhealth.py). There have been a few changes since you created your branch, which you should be able to bring in by [syncing your fork](https://help.github.com/articles/syncing-a-fork/) and merging those into your branch. One of those actually should help you see what output the spider is crawling to help you structure the tests.

**Comment by r-wei** [comment]
This issue is free for anyone to work on. Good first issue.

**Comment by r-wei** [comment]
resolved by #336
